tests/elf.rs

```diff
@@ -51,7 +51,12 @@ fn parse_text_section_size_lazy(base: &[u8]) -> Result<u64, &'static str> {
         SectionHeader::parse(base, header.e_shoff as usize, header.e_shnum as usize, ctx)
             .map_err(|_| "parse section headers error")?;
 
-    let strtab_idx = header.e_shstrndx as usize;
+    let strtab_idx = if header.e_shstrndx as u32 == SHN_XINDEX {
```

Comment:
weird i feel like this was added already, long time ago, surprised to see it now